File: Week 1 Game of life/Game_of_life (python) 2019-05-21/game_of_life.py
'''

   1. Any live cell with fewer than two live neighbours dies, as if caused by underpopulation.
   2. Any live cell with more than three live neighbours dies, as if by overcrowding.
   3. Any live cell with two or three live neighbours lives on to the next generation.
   4. Any dead cell with exactly three live neighbours becomes a live cell.

'''

class gameoflife():

    def __init__(self, _rows, _cols):
        self.n_rows = _rows
        self.n_cols = _cols
        self.grid = []
        self.game = 0
        self.creategrid()

    def creategrid(self):

        for x in range(self.n_rows):
            row = []
            for y in range(self.n_cols):
                row.append('.')

            self.grid.append(row)

    def printgrid(self):

        print("Generation #",self.game)
        for x in range(self.n_rows):
            for y in range(self.n_cols):
                print(self.grid[x][y],end="")
            print()

    def setgrid(self, x, y, val):
        self.grid[x][y]=val

    def play(self):

        for x in range(self.n_rows):
            for y in range(self.n_cols):
                cells = 0
                cells += self.getcell(x-1,y-1)
                cells += self.getcell(x,y-1)
                cells += self.getcell(x+1,y-1)
                cells += self.getcell(x-1,y)
                cells += self.getcell(x+1,y)
                cells += self.getcell(x-1,y+1)
                cells += self.getcell(x,y+1)
                cells += self.getcell(x+1,y+1)

                #if(cells == 3):
                    #print(self.grid[x][y])

                if(cells<2):
                    self.grid[x][y]='.'
                elif(cells>3):
                    self.grid[x][y]='.'
                elif(cells==3):
                    self.grid[x][y]='*'
        self.game+=1

    def getcell(self, x, y):
        if(x>=0 and x<self.n_rows and y>=0 and y<self.n_cols):
            if(self.grid[x][y]=='*'):
                return 1
        return 0

    

File: Week 1 Game of life/Game_of_life (C#) 2019-05-26/Game_of_life.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace ConsoleApp63
{
    class Game_of_life
    {
        public int rows;
        public int cols;
        char[,] grid;



        public Game_of_life(string dir)
        {
            this.build(dir);   
        }

        private void build(string dir)
        {
            string[] text_game = File.ReadAllLines(dir);
            this.rows = text_game[0][0]-'0';
            this.cols = text_game[0][2]-'0';
            this.grid = new char[this.rows, this.cols];

            for (int i = 0; i < this.rows; i++)
            {
                // S:\Desktop\game_of_life.txt

                string line = text_game[i+1];
                for(int j = 0; j < this.cols; j++)
                {
                    this.grid[i, j] = line[j];
                }
            }
        }


        public void printgrid()
        {
            Console.Write("\n");
            for (int i = 0; i < this.rows; i++)
            {
                for (int j = 0; j < this.cols; j++)
                {
                    Console.Write(this.grid[i,j]);
                }
                Console.Write("\n");
            }
        }

        public void play()
        {
            for (int i = 0; i < rows; i++)
            {
                for (int j = 0; j < cols; j++)
                {
                    int cells = 0;

                    cells += this.getcell(i - 1, j - 1);
                    cells += this.getcell(i, j - 1);
                    cells += this.getcell(i + 1, j - 1);
                    cells += this.getcell(i - 1, j);
                    cells += this.getcell(i + 1, j);
                    cells += this.getcell(i - 1, j + 1);
                    cells += this.getcell(i, j + 1);
                    cells += this.getcell(i + 1, j + 1);


                    if (cells < 2 || cells>3)
                        this.grid[i, j] = '.';
                    else if (cells == 3)
                        this.grid[i, j] = '*';
                }
            }
        }

        private int getcell(int x, int y)
        {
            if (x >= 0 && x < this.rows && y >= 0 && y < this.cols)
                return this.grid[x, y] == '*' ? 1 : 0;

            return 0;
        }


    }
}
